### PR TITLE
[embedded] Add a embedded Swift + typed throw test showcasing that we avoid existentials

### PR DESCRIPTION
**Pull request by kubamracek**
The actual compiler changes are taken from Doug's PR at <https://github.com/apple/swift/pull/68976/commits> (and I'll wait for that one to land first), I'm only adding the embedded testcase with this one.

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

File: test/embedded/throw-typed.swift [hunk 1]
```diff
@@ -0,0 +1,95 @@
+// RUN: %target-run-simple-swift(%S/Inputs/print.swift -enable-experimental-feature Embedded -enable-experimental-feature TypedThrows -parse-as-library -runtime-compatibility-version none -wmo -Xfrontend -disable-objc-interop) | %FileCheck %s
+
+// REQUIRES: executable_test
+// REQUIRES: optimized_stdlib
+// REQUIRES: VENDOR=apple
+// REQUIRES: OS=macosx
```
Comment (xedin) [inline test/embedded/throw-typed.swift:6]: I keep forgetting whether we still need `REQUIRES: asserts` when `-enable-experimental-feature` flag is used or not, could you please double-check?

Comment (kubamracek) [reply test/embedded/throw-typed.swift:6]: Some features need asserts, some don't (it's controlled in the Features.def file on a per feature basis). The ones used here don't need asserts.

**Comment by kubamracek** [comment]
@swift-ci please test

**Pull request merged**